render/render.js

```diff
@@ -628,65 +607,42 @@ module.exports = function($window) {
 			}
 		}
 		checkState(vnode, original)
-
+		var generation
 		// If we can, try to fast-path it and avoid all the overhead of awaiting
 		if (!mask) {
 			onremove(vnode)
-			removeChild(parent, vnode)
+			removeDOM(parent, vnode, undefined)
 		} else {
-			if (stateResult != null) {
-				var next = function () {
+			generation = currentRender
+			for (var dom of domFor(vnode)) delayedRemoval.set(dom, generation)
+			var finalizer = function(a, b) {
+				return function () {
 					// eslint-disable-next-line no-bitwise
-					if (mask & 1) { mask &= 2; if (!mask) reallyRemove() }
+					if (mask & a) { mask &= b; if (!mask) {
+						checkState(vnode, original)
+						onremove(vnode)
+						removeDOM(parent, vnode, generation)
+					} }
```

Comment:
This code is a bit rough to read, and the previous code only got away with it because it was all on one line total.

```suggestion
					/* eslint-disable no-bitwise */
					if (mask & a) {
						mask &= b
						if (!mask) {
							checkState(vnode, original)
							onremove(vnode)
							removeDOM(parent, vnode, generation)
						}
					}
					/* eslint-enable no-bitwise */
```

Reply:
Done